Hardware/Nic/Nic.cs

```diff
@@ -0,0 +1,148 @@
+﻿/*
+ 
+  This Source Code Form is subject to the terms of the Mozilla Public
+  License, v. 2.0. If a copy of the MPL was not distributed with this
+  file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ 
+  Copyright (C) 2012 Michael Möller <mmoeller@openhardwaremonitor.org>
+	
+*/
+
+using System;
+using System.Diagnostics;
+using System.Globalization;
+using System.Net.NetworkInformation;
+
+namespace OpenHardwareMonitor.Hardware.Nic
+{
+    internal class Nic : Hardware
+    {
+        private ISettings settings;
+        private Sensor connectionSpeed;
+        private Sensor dataUploaded;
+        private Sensor dataDownloaded;
+        private Sensor uploadSpeed;
+        private Sensor downloadSpeed;
+        private Sensor networkUtilization;
+        private Sensor totalDataDownloaded;
+        private Sensor totalDataUploaded;
+        private Sensor totalDataFlowed;
+        private NetworkInterface nic;
+        private int nicIndex;
+        private DateTime latesTime;
+        private DateTime presentBootTime;
+
+        private long bytesUploaded;
+        private long bytesDownloaded;
+        private long totalBytesDownloaded;
+        private long totalBytesUploaded;
+        private bool shouldTotalFlowUpdate = true;
+
+        public Nic(string name, ISettings Settings, int index, NicGroup nicGroup)
+          : base(name, new Identifier("NIC",index.ToString(CultureInfo.InvariantCulture)), Settings)
+        {
+            settings = Settings;
+            nicIndex = index;
+            nic = nicGroup.NicArr[index];
+            totalBytesDownloaded = Convert.ToInt64(settings.GetValue("TotalDownloadedBeforeLastBoot"+nic.Name, "-1"));
```

Comment:
I think this entire settings stuff should be removed here, the added value is very limited as it's only for monitored data while it is running. And it's causing a high number of writes when you update the data frequently.

Reply:
I dont know what you exactly  want to remove ? The Settings contructor ?

Reply:
No, settings is OK, but it's only used for custom names, not for storing the bytes like this here (which is rather pointless as it only monitors while OHM is running). I'll remove it 👍